### PR TITLE
Cl/item freeze

### LINKED ISSUE #40: Add Asset Freeze
As an asset owner, I want to be able to set my asset to unavailable manually at any time, so the asset stays in the system, but cannot be borrowed. This allows me to keep the asset to myself, e.g. to fix something or to handle errors in the lending process. I can revert this at any time. 

### Dependencies
- #8 
- #16 
- #30 

### Definition of Done
- [ ] Button on asset details page to freeze or defreeze the asset
- [ ] Frozen assets cannot be borrowed but stay in the system
- [ ] If item is currently borrowed and owner freezes, it will be frozen when returned

### PR DESCRIPTION
**Pull request by TimRiedel**
closes #40

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -236,6 +236,16 @@ def leave_waitlist
     end
   end
 
+  # PATCH
+  def toggle_status
+    @item.toggle_status
+
+    respond_to do |format|
```
Comment (konrad-gerlach) [inline app/controllers/items_controller.rb:243]: Save here and if saving fails return error message; Also make sure to check rights for toggling status.

**Review by konrad-gerlach — CHANGES_REQUESTED** [review]
Schicke Implementation! Eine Kleinigkeit sollte noch im Controller angepasst werden.

**Review by DefineDan — CHANGES_REQUESTED** [review]
Looks and feels very nice overall. One thing: When the item is overdue, then the item freeze causes some unintuitive behavior. The second info box should rather stay the same, saying how to return the item and not that the item is unavailable.

item freeze 1
item freeze 2

File: app/controllers/items_controller.rb (hunk 1 above)
Comment (johannaschlimme) [reply app/controllers/items_controller.rb:252]: Thanks for this feedback. Please have a look at the new commit refactor: add save and permission check in toggle_status](https://github.com/hpi-swt2/bookkeeper-portal-red/pull/367/commits/8d8ba7853134fce505a85e27e46eb9d0956b2062) and let us know if there are still some issues

**Review by konrad-gerlach — CHANGES_REQUESTED** [review]
Looks good for now but I would like to have another look once the changes requested by @DefineDan have been added.

**Review by DefineDan — APPROVED** [review]

**Review by konrad-gerlach — APPROVED** [review]
Sieht schön aus! Gute Tests und cleane Implementierung.

**Pull request merged**